reference_test.go

```diff
@@ -337,7 +344,7 @@ func TestWithNameFailure(t *testing.T) {
 	}{
 		{
 			input: "",
-			err:   ErrNameEmpty,
+			err:   ErrReferenceInvalidFormat,
```

Comment:
Hm.. why did this one change? Not sure if that's expected 🤔

Reply:
It's already returning `ErrReferenceInvalidFormat` on main, without the new changes. Should I rather make the test pass by changing the behaviour instead?

Reply:
Sorry for the confusion about making the test pass. The test would pass, but it would return `ErrReferenceInvalidFormat` instead of `ErrNameEmpty` which is the expectation. The test will not fail as we are only asserting on main that there is an error and not asserting which error it is.

If you change the assertion from:
```
if err == nil {
    t.Errorf("no error parsing name. expected: %s", tc.err)
}
```
to: 
```
if !errors.Is(err, tc.err) {
    t.Errorf("unexpected error parsing name. expected: %s, got: %s", tc.err, err)
}
```
to check the actual error that is returned is equal to the expectation in the test, the test will fail. So yes, the test is passing but we are not asserting correctly what is expected in the test input